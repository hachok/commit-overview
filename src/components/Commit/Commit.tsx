import React, {Fragment} from 'react';
import {
    Avatar,
    Divider,
    ListItem,
    ListItemAvatar,
    ListItemText, makeStyles,
    Typography
} from '@material-ui/core';
import {format} from 'date-fns';
import {CommitData} from 'src/types';

interface Props {
    info: CommitData

    onClick?(commitId: string): void
}

const useStyles = makeStyles(() => ({
    wrapper: {
        cursor: 'pointer',
    },
    inline: {
        display: 'inline',
    },
    listItem: {
        height: '45px',
        overflow: 'hidden',
        '& p': {
            display: 'inline-block',
            width: '100%',
            whiteSpace: 'nowrap',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
        },
    }
}));

const Commit: React.FC<Props> = ({info, onClick}) => {
    const classes = useStyles();

    const handleClick = () => {
        onClick && onClick(info.sha);
    }

    return (
        <div data-testid="commit-element" className={classes.wrapper} onClick={handleClick}>
            <ListItem alignItems="flex-start">
                <ListItemAvatar>
                    <Avatar alt="Avatar" src={info.author.avatar_url}/>
                </ListItemAvatar>
                <ListItemText
                    primary={info.commit.author.name}
                    className={classes.listItem}
                    secondary={
                        <Fragment>
                            <Typography
                                component="span"
                                variant="body2"
                                className={classes.inline}
                                color="textPrimary"
                            >
                                {format(Date.parse(info.commit.author.date), 'dd-MM-yyyy')}
                            </Typography>
                            {' - '}
                            {info.commit.message}
                        </Fragment>
                    }
                />
            </ListItem>
            <Divider variant="inset"/>
        </div>
    );
}
export default Commit;