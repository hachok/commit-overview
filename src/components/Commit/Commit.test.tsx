import Commit from 'src/components/Commit/Commit';
import {render, fireEvent} from '@testing-library/react';

const mockData = {
    sha: '1234',
    commit: {
        author: {
            name: 'name',
            email: 'email@gmail.com',
            date: '2021-03-07T23:43:29Z'
        }
    },
    author: {
        avatar_url: 'https://avatars.githubusercontent.com/u/2440089?v=4'
    }
};

test('it should render component correctly', async () => {
    const rendered = render(<Commit info={mockData} onClick={jest.fn}/>);

    expect(rendered).toMatchSnapshot();
})

test('it should call onClick function', async () => {
    const mockJestFn = jest.fn();
    const rendered = render(<Commit info={mockData} onClick={mockJestFn}/>);

    const {getByTestId} = rendered;

    const commitElement = getByTestId('commit-element');

    fireEvent.click(commitElement)

    expect(mockJestFn).toHaveBeenCalled();
})
