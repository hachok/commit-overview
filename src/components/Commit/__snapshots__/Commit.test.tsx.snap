// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-wrapper-1"
        data-testid="commit-element"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="Avatar"
                class="MuiAvatar-img"
                src="https://avatars.githubusercontent.com/u/2440089?v=4"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItem-3 MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              name
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              <span
                class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                08-03-2021
              </span>
               - 
            </p>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-inset"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-wrapper-1"
      data-testid="commit-element"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="Avatar"
              class="MuiAvatar-img"
              src="https://avatars.githubusercontent.com/u/2440089?v=4"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root makeStyles-listItem-3 MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            name
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              08-03-2021
            </span>
             - 
          </p>
        </div>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-inset"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
